flink-runtime/src/main/java/org/apache/flink/runtime/state/TaskLocalStateStoreImpl.java

```diff
@@ -300,6 +291,32 @@ private void deleteDirectory(File directory) throws IOException {
 		}
 	}
 
+	/**
+	 * Pruning the useless checkpoints.
+	 */
+	private void pruneCheckpoints(long checkpointID, boolean breakTheIteration) {
+
```

Comment:
I suggest to add an assert that the thread holds `lock` and document that this method should be called only when holding the lock.

Reply:
👍